lib/fastlane/actions/xcodebuild.rb

```diff
@@ -0,0 +1,129 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      XCODEBUILD_ARCHIVE = :XCODEBUILD_ARCHIVE
+    end
+
+    # xcodebuild man page:
+    # https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man1/xcodebuild.1.html
+
+    class XcodebuildAction
+      ARGS_MAP = {
+        # actions
+        analyze: "analyze",
+        archive: "archive",
+        build: "build",
+        clean: "clean",
+        install: "install",
+        installsrc: "installsrc",
+        test: "test",
+
+        # parameters
+        arch: "-arch",
+        alltargets: "-alltargets",
+        archive_path: "-archivePath",
+        configuration: "-configuration",
+        derivedDataPath: "-derivedDataPath",
+        destination: "-destination",
+        export_archive: "-exportArchive",
+        export_format: "-exportFormat",
+        export_installer_identity: "-exportInstallerIdentity",
+        export_path: "-exportPath",
+        export_profile: "-exportProvisioningProfile",
+        export_signing_identity: "-exportSigningIdentity",
+        export_with_original_signing_identity: "-exportWithOriginalSigningIdentity",
+        project: "-project",
+        result_bundle_path: "-resultBundlePath",
+        scheme: "-scheme",
+        sdk: "-sdk",
+        skip_unavailable_actions: "-skipUnavailableActions",
+        target: "-target",
+        workspace: "-workspace",
+        xcconfig: "-xcconfig"
+      }
+
+      def self.run(params)
+        unless Helper.test?
+          raise "xcodebuild not installed".red if `which xcodebuild`.length == 0
+        end
+
+        # The args we will build with
+        cli_args = Array[]
+
+        # Supported ENV vars
+        build_path = ENV["BUILD_PATH"]
+        scheme     = ENV["SCHEME"]
+        workspace  = ENV["WORKSPACE"]
+        project    = ENV["PROJECT"]
```

Comment:
@dtrenz Are there ENV variables required? I see `build_path` and `scheme` being used below (https://github.com/KrauseFx/fastlane/pull/104/files#diff-6068f34d3ade6167ee700963b4ace891R78) but only possibly being instantiated here